hazelcast/src/main/java/com/hazelcast/query/impl/OrderedIndexStore.java

```diff
@@ -110,6 +115,81 @@ public boolean canEvaluate(Class<? extends Predicate> predicateClass) {
         throw new UnsupportedOperationException();
     }
 
+    @Override
+    public Iterator<QueryableEntry> getRecordIterator() {
+        Iterator<QueryableEntry> iterator = new IndexEntryCompositeIterator(recordMap.values().iterator());
+        Iterator<QueryableEntry> nullIterator = recordsWithNullValue.values().iterator();
+
+        return new FlatCompositeIterator<>(Arrays.asList(iterator, nullIterator).iterator());
```

Comment:
I'd return null values first because `NULL` is less than any other key.

Reply:
Moved `NULL` values to the beginning of the data stream.